File: exemplos/capitulo06/enum.c
#include <stdio.h>

typedef enum Cores {vermelho, azul, verde, amarelo, preto} Cores;

int main(int argc, char *argv) {
    Cores minhaCor = azul;
    Cores suaCor = amarelo;

    // Aritmética
    minhaCor++;

    if (suaCor > minhaCor) {
        printf("Posso fazer comparações de ordenação.\n");
    }

    if (minhaCor == verde) {
        printf("Posso fazer aritmética de cores.\n");
    }

    // Outro exemplo
    minhaCor = suaCor + 1;
    if (minhaCor == preto) {
        printf("Posso fazer aritmética de cores.\n");
    }

    // Fora da faixa, sem problemas.
    suaCor = 5;
    
    printf("Valores inteiros: %d %d\n", minhaCor, suaCor);

    return 0;
}

File: exemplos/capitulo06/Enum.java
public class Enum {
    public enum Cores{
        Vermelho,
        Azul,
        Verde,
        Amarelo,
        Preto
    }
    public static void main(String[] args) {
        for (Cores estacao : Cores.values()) {
            System.out.println(estacao);
        }

        // É preciso especificar os valores como sendo da enumeração.
        Cores minhaCor = Cores.Azul;

        // É possível?
        // minhaCor = 1;
        System.out.println("Minha Cor é: " + minhaCor);
        System.out.println("Minha Cor é: " + minhaCor.ordinal());

    }
 }